proto/viam/app/data/v1/data.proto

```diff
@@ -226,6 +229,20 @@ message TabularDataByMQLResponse {
   reserved "data";
 }
 
+// GetLatestTabularDataRequest requests the most recent tabular data captured from the specified data source.
+message GetLatestTabularDataRequest {
+  string part_id = 1;
+  string resource_name = 2;
+  string method_name = 3;
+}
+
+// GetLatestTabularDataResponse provides the data, time synced, and time captured of most recent tabular data captured from the requested data source.
+message GetLatestTabularDataResponse {
+  google.protobuf.Timestamp time_captured = 1;
+  google.protobuf.Timestamp time_synced = 2;
+  google.protobuf.Struct payload = 3;
```

Comment:
Thinking about the type we return here... perhaps we can return a byte array (so `repeated bytes`) instead of a Struct, similar to what we've done with the ByMQL/BySQL response type. This would allow for more expressive data types in the payload (for example, users would be able to interact with a returned date directly instead of first having to parse it from a string).

@dmhilly I know there was some discussion recently around the raw_data vs data return type for the ByQuery endpoints, what was the resolution on that? Okay to return a byte array here instead of a struct?

Reply:
Good question! I was thinking about this when I reviewed the scope too... my thought is that it's okay and preferred to return it as a protobuf.Struct, the reasoning being that to get it into Viam Cloud, it would have to have been uploaded as a protobuf.Struct. DataCaptureUpload takes 1+ SensorDatas, which each contains some metadata + either raw bytes or a protobuf.Struct, see [here](https://github.com/viamrobotics/api/blob/cdaf1d58f503c0c373a3a3e3d58d8fe2fc83763a/proto/viam/app/datasync/v1/data_sync.proto#L89). If the mime_type in the request is tabular but the SensorData data does not contain a struct, we get an error [here](https://github.com/viamrobotics/app/blob/1bab5d36083d69e2fe685442f141844f5a867be2/datasync/tabular_data_upload.go#L127).

For TabularDataByMQL, the query is across both the data and the metadata, and since there are datetimes in the metadata (the timestamps), these fields (or any fields constructed by the query that aggregates/modifies/defines new fields) had the wrong data type. The query could also theoretically define other fields with BSON data types that are not expressible in proto.

Reply:
Gotcha, thanks for that explanation - with all that in mind keeping this as a protobuf.Struct makes sense to me as well :)